packages/web-app-files/tests/unit/views/LocationPicker.spec.js

```diff
@@ -0,0 +1,459 @@
+              query: {
+                resource: [route.query.resource]
+              }
+            }
+          })
+          expect(breadcrumbItems[2]).toMatchObject({
+            index: 2,
+            text: 'child'
+          })
+          expect(breadcrumbItems[2].to).toBeUndefined()
+        }
+      )
+    })
+    describe('cancel button', () => {
+      it('should show the location picker cancel button', () => {
+        const wrapper = getShallowWrapper()
+        const cancelButton = wrapper.find(selectors.cancelButton)
+
+        expect(cancelButton.exists()).toBeTruthy()
+        expect(cancelButton.find(translateStub).exists()).toBeTruthy()
+        expect(cancelButton).toMatchSnapshot()
+      })
+      it('should call "leaveLocationPicker" method when clicked', async () => {
+        const spyLeaveLocationPicker = jest.spyOn(LocationPicker.methods, 'leaveLocationPicker')
+        const spyRouterPush = jest.spyOn(router, 'push')
+
+        const wrapper = getMountedWrapper()
+        const cancelButton = wrapper.find(selectors.cancelButton)
+
+        await cancelButton.trigger('click')
+
+        expect(spyLeaveLocationPicker).toHaveBeenCalledTimes(1)
+        expect(spyLeaveLocationPicker).toHaveBeenCalledWith('/some/item')
+        expect(spyRouterPush).toHaveBeenCalledTimes(1)
+        expect(spyRouterPush).toHaveBeenCalledWith({
+          name: 'files-personal',
+          params: { item: '/some/item' }
+        })
+      })
+    })
+    describe('should show the location picker confirm button', () => {
+      it('should be disabled if no current folder', () => {
+        const wrapper = getShallowWrapper()
+
+        const confirmButton = wrapper.find(selectors.confirmButton)
+
+        expect(confirmButton.exists()).toBeTruthy()
+        expect(confirmButton.attributes().disabled).toBe('true')
+      })
+      it('should be disabled if current folder does not have "canCreate" property', () => {
+        const route = getRoute()
+        const store = createStore({ currentFolder: { canCreate: jest.fn(() => false) } })
+        const wrapper = getShallowWrapper({ store: store, $route: route })
+
+        const confirmButton = wrapper.find(selectors.confirmButton)
+
+        expect(confirmButton.exists()).toBeTruthy()
+        expect(confirmButton.attributes().disabled).toBe('true')
+      })
+      it.each(['move', 'copy', 'anything else'])(
+        'should set button text according to current action',
+        action => {
+          const wrapper = getShallowWrapper({
+            store: createStore({ currentFolder: { canCreate: jest.fn(() => true) } }),
+            $route: getRoute({ action: action })
+          })
+
+          const confirmButton = wrapper.find(selectors.confirmButton)
+
+          expect(confirmButton.exists()).toBeTruthy()
+          expect(confirmButton.attributes().disabled).toBeUndefined()
+          expect(confirmButton).toMatchSnapshot()
```

Comment:
Check pls directly the text instead of snapshot

Reply:
fixed